modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -661,6 +664,21 @@ public UserAccount patchUserAccount(
 		return _toUserAccount(user);
 	}
 
+	@Override
+	public UserAccount patchUserAccountByExternalReferenceCode(
+			String externalReferenceCode, UserAccount userAccount)
+		throws Exception {
+
+		User user = _userService.fetchUserByExternalReferenceCode(
+			contextCompany.getCompanyId(), externalReferenceCode);
+
+		if (user == null) {
+			throw new NoSuchUserException();
```

Comment:
Exception message is missing:
throw new NoSuchUserException(
				"No user exists with external reference code " +
					externalReferenceCode);

But recently, we used a new approach and implemented getXXXXXByExternalReferenceCode which makes the null check redundant so it can be removed.
For that you need to implement the getUserByExternalReferenceCode() method in UserServiceImpl and UserServiceLocalImpl (I could not see it there either).

@smottal Please confirm one of the above.

Reply:
Yes, now in master there should be _userService.getUserByExternalReferenceCode that throws already the exception if is null, so here we can avoid doing the fatch + throw the error manually